Basic-Car-Maintenance/Shared/Dashboard/ViewModels/DashboardViewModel.swift

```diff
@@ -66,6 +68,7 @@ class DashboardViewModel {
     }
     
     func getMaintenanceEvents() async {
+        loadingState = .isLoading
```

Comment:
I haven't tested this yet, but what happens when an error is thrown, is this forever "loading?"

Reply:
Yes, the loading spinner would continue to show.

Reply:
Does the alert show?

Reply:
I added an error state to the `LoadingState` enum and created a new error alert to show.